### PR TITLE
[BugFix] SliceMutableIndex's snapshot size is calculated incorrectly

### LINKED ISSUE #10666: [persistent index] BE cannot startup successfully and continously crash after upgrade
### Steps to reproduce the behavior (Required)
BE cannot startup successfully and continously crash after upgrade from branch-2.4 to branch-2.4:
```
Core was generated by `/home/disk2/sr/stability_master_01/be-7063d832-9520-477b-ba14-d391fc64b9b2/lib/'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000000022cc160 in starrocks::Slice::Slice (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x7fd3daa3d7f0) at /root/starrocks/be/src/storage/persistent_index.cpp:1142
1142	/root/starrocks/be/src/storage/persistent_index.cpp: No such file or directory.
[Current thread is 1 (Thread 0x7fdae93d3700 (LWP 57067))]
(gdb) bt
Python Exception <type 'exceptions.ImportError'> No module named gdb.frames:
#0  0x00000000022cc160 in starrocks::Slice::Slice (this=0x7fd3daa3d7f0) at /root/starrocks/be/src/storage/persistent_index.cpp:1142
#1  starrocks::SliceMutableIndex::load (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0xa6beaac0, offset=@0x7fdae93cc540: 2237, file=...) at /root/starrocks/be/src/storage/persistent_index.cpp:1142
#2  0x000000000222ad7e in starrocks::ShardByLengthMutableIndex::load (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0xa6c16000, meta=...) at /root/starrocks/be/src/storage/persistent_index.cpp:1798
#3  0x00000000022420dc in starrocks::PersistentIndex::_load (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x9ffb4d00, index_meta=...) at /root/starrocks/be/src/storage/persistent_index.cpp:2281
#4  0x0000000002243597 in starrocks::PersistentIndex::load (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0x9ffb4d00, index_meta=...) at /root/starrocks/be/src/storage/persistent_index.cpp:2263
#5  0x0000000002244f06 in starrocks::PersistentIndex::load_from_tablet (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=this@entry=0x9ffb4d00, tablet=tablet@entry=0xa6a77c00)
    at /root/starrocks/be/src/storage/persistent_index.cpp:2432
#6  0x0000000001f947ec in starrocks::PrimaryIndex::_do_load (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0xbd67df0, tablet=0xa6a77c00) at /root/starrocks/be/src/storage/primary_index.cpp:1028
#7  0x0000000001f9556f in starrocks::PrimaryIndex::load (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=this@entry=0xbd67df0, tablet=0xa6a77c00) at /root/starrocks/be/src/storage/primary_index.cpp:937
#8  0x0000000001e1aa36 in starrocks::TabletUpdates::_apply_compaction_commit (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0xa6b74000, version_info=...) at /root/starrocks/be/src/storage/tablet_updates.cpp:1293
#9  0x0000000001e1cc8d in starrocks::TabletUpdates::do_apply (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0xa6b74000) at /root/starrocks/be/src/storage/tablet_updates.cpp:731
#10 0x0000000002612275 in starrocks::ThreadPool::dispatch_thread (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
this=0xb33d880) at /root/starrocks/be/src/util/threadpool.cpp:209
#11 0x000000000260d6aa in std::function<void ()>::operator()() const (this=0xbd67cf8) at /usr/include/c++/10.3.0/bits/std_function.h:248
#12 starrocks::Thread::supervise_thread (Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
arg=0xbd67ce0) at /root/starrocks/be/src/util/thread.cpp:326
#13 0x00007fdb0120eea5 in start_thread () from /lib64/libpthread.so.0
Python Exception <type 'exceptions.NameError'> Installation error: gdb._execute_unwinders function is missing:
#14 0x00007fdb00829b0d in clone () from /lib64/libc.so.6
(gdb)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| 2.4.0-0901 133fd49 |
+--------------------+
1 row in set (0.12 sec)
```

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10666

## Problem Summary(Required) ：

The snapshot size of SliceMutableIndex is incorrect, so we may get an unexpected error when we try to build a persistent index from existing files.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by sevev** [comment]
run starrocks_admit_test

File: be/src/storage/persistent_index.cpp [hunk 1]
```diff
@@ -1063,9 +1063,7 @@ class SliceMutableIndex : public MutableIndex {
     }
 
     // return the dump file size if dump _set into a new file
-    // If _set is empty, the real snapshot file will only wite a size_(type is size_t) into file. So we
-    // will use `sizeof(size_t)` as return value.
-    size_t dump_bound() { return sizeof(size_t) + _total_kv_pairs_usage; }
+    size_t dump_bound() { return sizeof(size_t) + _total_kv_pairs_usage + sizeof(size_t) * size(); }
```
Comment (chaoyli) [inline be/src/storage/persistent_index.cpp:1066]:
```suggestion
    size_t dump_bound() { return (sizeof(size_t) + 1) * size() + _total_kv_pairs_usage; }
```

Comment (sevev) [reply be/src/storage/persistent_index.cpp]: > 

done, update in https://github.com/StarRocks/starrocks/pull/10772/commits/c20ce64b14bcf7cf3065ced2c80bb04e6b35f853

**Review by decster — APPROVED** [review]

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
@mergify backport branch-2.4

**Pull request merged**